docs/DeveloperGuide.md

```diff
@@ -393,6 +393,61 @@ Two possible behaviours were considered when designing the recall command featur
   * Pros: Easy to implement.
   * Cons: The user looses any partially typed commands.
 
+### Attendance Model
+This section explains the design considerations of the `Attendance` model.
+
+#### Implementation
+The `Attendance`, `AttendanceRecord` and `AttendanceRecordList` models are implemented as follows:
+
+![AttendanceClassDiagram](images/AttendanceClassDiagram.png)
+
+`Attendance` encapsulates the information of a particular student's attendance in a specific lesson he/she has attended.
+`AttendanceRecord` refers to the actual attendance roster of a particular lesson event.
+It contains the `Attendances` of all `Students` who have attended the particular lesson.
```

Comment:
```suggestion
It contains the `Attendance`s of all `Students` who have attended the particular lesson.
```
Or perhaps we can rephrase the sentence since the above does not get rendered as nicely?